### PR TITLE
Implement Gated Recurrent Unit (GRU)

### LINKED ISSUE #92: Implement Gated Recurrent Unit
Suggested by @chrisemoody at twitter. It is desirable to implement GRU unit to cover various kind of architecture.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #92

### CODE REVIEW

**Comment by unnonouno** [comment]
I implemented GRU function, but a test code which is commented out is not passed.

File: tests/chainer_tests/functions_tests/activation_tests/test_gru.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def check_backward(self, h_data, x_data, y_grad):
+        h = chainer.Variable(h_data)
+        x = chainer.Variable(x_data)
+        y = self.func(h, x)
+
+        y.grad = y_grad
+        y.backward()
+
+        f = lambda: _gru(self.func, h.data, x.data)
```
Comment (beam2d) [inline tests/chainer_tests/functions_tests/activation_tests/test_gru.py]: It should be `f = lambda: (_gru(self.func, h.data, x.data),)`.

**Comment by beam2d** [comment]
Would you write a simple docstring, in particular with a link to the paper?

**Comment by beam2d** [comment]
LGTM!

**Pull request merged**